pkg/requester/jobtransform/docker_image_digest.go

```diff
@@ -0,0 +1,46 @@
+package jobtransform
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/rs/zerolog/log"
+)
+
+func DockerImageDigest() Transformer {
+	return func(ctx context.Context, j *model.Job) (modified bool, err error) {
+		client, err := docker.NewDockerClient()
+		if err != nil || !client.IsInstalled(ctx) {
+			// Return a noop if docker is not installed as it means we
+			// won't be able to find digests for images in the requester
+			return false, nil
+		}
+
+		if j.Spec.Engine != model.EngineDocker {
+			return false, nil
+		}
+
+		image, err := docker.NewImageID(j.Spec.Docker.Image)
+		if err != nil {
+			return false, nil
+		}
+
+		resolver := docker.NewImageResolver(image)
+		err = resolver.Resolve(ctx, client.ImageDistribution, docker.DockerTagCache)
+		if err != nil {
+			log.Ctx(ctx).Debug().
+				Str("Image", image.String()).
```

Comment:
You can use `Stringer` here to log an item with a String method, the advantage being that is the String method is expensive it won't be called unless the log message will actually be sent. May not be relevant here depending on the String method, but I thought it's worth mentioning for the future.

Reply:
TIL, thanks!